models/iosxr/vrf/iosxr_vrf.yaml

```diff
@@ -0,0 +1,124 @@
+module: iosxr_vrf
+short_description: Resource module to manage global VRF configuration.
+description:
+  - This module manages VRF definitions on Cisco IOS_XR devices. It enables playbooks to handle either individual VRFs or the complete VRF collection. It also permits removing non_explicitly stated VRF definitions from the setup.
+version_added: 7.2.0
+author: Ruchi Pakhle (@Ruchip16)
+notes:
+  - Tested against Cisco IOS_XR Version 17.3 on CML.
+  - This module works with connection C(network_cli). See L(the IOS_XR Platform Options,../network/user_guide/platform_iosxr.html)
+  - For more information on using Ansible to manage network devices see the :ref:`Ansible Network Guide <network_guide>`
+  - For more information on using Ansible to manage Cisco devices see the `Cisco integration page <https://www.ansible.com/integrations/networks/cisco>`_.
+options:
+  config:
+    vrfs:
+      description: A list of device configurations for VRF.
+      type: dict
+      suboptions:
+        address_families:
+          description: Enable address family and enter its config mode - AFI/SAFI configuration
+          type: list
+          elements: dict
+          suboptions:
+            afi:
+              description: Address Family Identifier (AFI)
+              type: str
+              choices: ['ipv4', 'ipv6']
+            safi:
+              description: Address Family modifier
+              type: str
+              choices: [ 'flowspec', 'multicast', 'unicast']
+            export:
+              description: VRF export
+              type: dict
+            import:
+              description: VRF import
+              type: dict
```

Comment:
This seems to be a dict, but has no suboptions.

Reply:
thats because for the suboptions ashwini said that if they are repetitive i can just write them only once in this way, import and export had some repetitive suboptions like` route-policy` and `route-target `is common in both so i instead wrote them only once below, should i put them under suboptions?

Reply:
I think what @ashwini-mhatre suggested was to use YAML anchoring to reuse existing key-value pairs. As an example:

https://github.com/ansible-collections/cisco.nxos/blob/main/plugins/modules/nxos_bgp_global.py#L93-L95

https://github.com/ansible-collections/cisco.nxos/blob/main/plugins/modules/nxos_bgp_global.py#L488